pkg/providers/instanceprofile/instanceprofile.go

```diff
@@ -204,14 +226,14 @@ func (p *DefaultProvider) ListNodeClassProfiles(ctx context.Context, nodeClass *
 }
 
 func (p *DefaultProvider) IsProtected(profileName string) bool {
-	_, exists := p.protectedProfiles.Get(profileName)
+	_, exists := p.protectedProfileCache.Get(profileName)
 	return exists
 }
 
 func (p *DefaultProvider) SetProtectedState(profileName string, protected bool) {
```

Comment:
Why does this function even exist? Seems like a very loose wrapper around just calling the cache directly